tuya.cpp

```diff
@@ -474,256 +528,421 @@ void DeRestPluginPrivate::handleTuyaClusterIndication(const deCONZ::ApsDataIndic
+                        ResourceItem *item = sensorNode->item(RConfigMelody);
 
-                    if (item && item->toString() != mode && data == 0) // Only change if off
-                    {
-                        item->setValue(mode);
-                        enqueueEvent(Event(RSensors, RConfigMode, sensorNode->id(), item));
+                        if (item && item->toNumber() != melody)
+                        {
+                            item->setValue(melody);
+                            enqueueEvent(Event(RSensors, RConfigMelody, sensorNode->id(), item));
+                            update = true;
+                        }
+                        
                     }
-                }
-                break;
-                case 0x0168: // Alarm
-                {
-                    bool alarm = false;
-                    if (data == 1) { alarm = true; }
+                    break;
+                    case 0x0474 : // volume
+                    {
+                        quint8 volume = (static_cast<qint8>(data & 0xFF));
 
-                    ResourceItem *item = sensorNode->item(RStateAlarm);
+                        ResourceItem *item = sensorNode->item(RConfigVolume);
 
-                    if (item && item->toBool() != alarm)
-                    {
-                        item->setValue(alarm);
-                        Event e(RSensors, RStateAlarm, sensorNode->id(), item);
-                        enqueueEvent(e);
+                        if (item && item->toNumber() != volume)
+                        {
+                            item->setValue(volume);
+                            enqueueEvent(Event(RSensors, RConfigVolume, sensorNode->id(), item));
+                            update = true;
+                        }
                     }
+                    break;
+                    default:
+                    break;
                 }
-                break;
-                case 0x016A: // Away mode
-                {
-                    //bool away = false;
-                    //if (data == 1) { away = true; }
-                }
-                break;
-                case 0x016c: // manual / auto
+            }
+            else
+            {
+                // Generic part
+                switch (dp)
                 {
-                    QString mode;
-                    if      (data == 0) { mode = QLatin1String("heat"); } // was "manu"
-                    else if (data == 1) { mode = QLatin1String("auto"); } // back to "auto"
-                    else
+                    case 0x0068: // window open information
                     {
-                        return;
-                    }
+                        quint8 valve = static_cast<quint8>(dp & 0xFF);
+                        quint8 temperature = static_cast<quint8>((dp >> 8) & 0xFF);
+                        quint8 minute = static_cast<quint8>((dp >> 16) & 0xFF); // TODO, U16 >> 16 ... this is always 0?!
```

Comment:
Not used for the moment, ATM we can set them, but the return is not used.
It s more for memory, if one time I need to use them

Reply:
In that case I wouldn't add it to the PR or at least comment it out with a note what it is and why it isn't used.
In general I'm *very* against adding dead code since usually it stays there for years with nobody having a clue why it's there and everybody afraid of removing it ;)